### PR TITLE
Update dependencies to latest versions

### PR DESCRIPTION
**Pull request by merlimat**
Fixes #<xyz>

Master Issue: #<xyz>

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [ ] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by BewareMyPower — COMMENTED** [review]
The download link of protobuf 25.1 is different from old versions.

https://github.com/protocolbuffers/protobuf/releases/tag/v25.1

**Comment by merlimat** [comment]
> The download link of protobuf 25.1 is different from old versions.

I actually rolled back the protobuf update, it's too complex. They are now requiring another dependency to be installed https://abseil.io/

File: dependencies.yaml [hunk 1]
```diff
@@ -17,11 +17,11 @@
...
+cmake: 3.28.3
 protobuf: 3.20.0
-zlib: 1.2.12
-zstd: 1.5.2
-snappy: 1.1.9
-openssl: 1.1.1v
-curl: 8.4.0
+zlib: 1.3.1
+zstd: 1.5.5
+snappy: 1.1.10
```
Comment (BewareMyPower) [inline dependencies.yaml:25]: Snappy 1.1.0 seems not work with the latest Apple Clang

```
snappy-1.1.10/snappy.cc:1293:37: error: comparison of integers of different signs: 'unsigned long' and 'ptrdiff_t' (aka 'long') [-Werror,-Wsign-compare]
```

We might need some fix on the build script.

**Comment by BewareMyPower** [comment]
I continued this work in another PR: https://github.com/apache/pulsar-client-cpp/pull/404. So close it for now.

**Pull request closed without merging**